lib/ElementHandle.js

```diff
@@ -260,6 +260,21 @@ class ElementHandle extends JSHandle {
     return null;
   }
 
+  /**
+   * @param {string} selector
+   * @param {function()} pageFunction
+   * @param {!Array<*>} args
+   * @return {!Promise<(!Object|undefined)>}
+   */
+  async $eval(selector, pageFunction, ...args) {
+    const elementHandle = await this.$(selector);
```

Comment:
once you have this in place, you can re-use this inside `Frame.$eval`, similarly to how it's done for `Frame.$`.